docs_src/design/adr/system-management/0006-Metrics-ControlPlaneEvents.md

```diff
@@ -0,0 +1,226 @@
+
+### Requirements
+
+- Services will push specified metrics collected for that service to a specified (by configuration) message endpoint (MQTT topic, 0MQ topic or other EdgeX supported messaging implementation)
+  - Use and configuration to target a secure endpoint (MQTTS) must be provided.  As exemplified by application services, there must be the option to use "insecure secrets" or Vault-backed security services when a secure endpoint is desired.
+- Services will push all CPE detected for that service to a specified (by configuration) message endpoint.
+  - The endpoint for metric and CPE may be the same or different at the user's discretion.
+  - Use and configuration to target a secure endpoint (MQTTS) must be provided.  As exemplified by application services, there must be the option to use "insecure secrets" or Vault-backed security services when a secure endpoint is desired.
+- All services must document what metrics and CPE they offer.
+- Services must provide REST endpoints that respond with what metrics and CPE the service offers (and whether that metric or CPE is currently `on` or `off`).
+  - Proposed endpoint format:  /api/v2/metrics and /api/v2/cpe
+- Services will have configuration which allows EdgeX system managers to select which metrics and CPE are turned `on` or `off`.
+  - When a metric or CPE is turned `off` the service does not collect the metric or monitor for the CPE.  When a metric is turned `on` the service collects and sends the metric to the designated endpoint.  When a CPE is turned `on` the service monitors for the event and sends any detected CPE to the designated endpoint.
+  - All metrics collection and CPE monitoring and sending is `off` by default.
+  - Metrics collection must be harvested on some appointed schedule.  This could be designated by configuration in the schedule service or done in a way similar to auto events in device services.  *Do we want to dictate this or allow services to implement as they see fit?*
+  - CPE is triggered by some happening in the service that the service monitors for and triggers the publishing of a new CPE message when the event occurs.
+
+### Requested Metrics
+
+The following is a list of example metrics and CPE requested by the EdgeX community and adopters for various service areas.  Number metrics would generally be processed in some interval (last 1/5/15 minutes or other defined interval).
+
+#### General
+
+The following metrics apply to all (or most) services.
+
+- Service uptime (time since last service boot)
+- Cumulative number API requests succeeded / failed / invalid (2xx vs 5xx vs 4xx)
+- Avg response time (in milliseconds or appropriate unit of measure) on APIs
+- Avg and Max request size
+
+#### Core/Supporting
+
+- Latency (measure of time) an event takes to get through core data
+- Latency (measure of time) a command request takes to get to a device service
+- Indication of health – that events are being processed during a configurable period

```

Comment:
Is this more of a control plane event, i.e. haven't received an event in configured time interval?